### PR TITLE
Allow full read from system database

### PR DESCRIPTION
**Pull request by pufit**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `allow_full_read_from_system_databases` user setting, which allows reading all rows from system tables even without grants to related tables. This setting doesn't grant read on `system.*` by itself!

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

Example of usage:
```
<users>
  <user_name>
    <allow_full_read_from_system_databases>1</allow_full_read_from_system_databases>
    <allow_databases>
        <database>system</database>
    </allow_databases>
    
    <!-- ... -->
  </user_name>
</users>
```

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/System/StorageSystemColumns.cpp [hunk 1]
```diff
@@ -90,7 +91,8 @@ class ColumnsSource : public ISource
         MutableColumns res_columns = getPort().getHeader().cloneEmptyColumns();
         size_t rows_count = 0;
 
-        const bool check_access_for_tables = !access->isGranted(AccessType::SHOW_COLUMNS);
+        const auto user = access->getUser();
+        const bool check_access_for_tables = !access->isGranted(AccessType::SHOW_TABLES) && !user->allow_full_read_from_system_databases;
```
[the remark this answers is not in the document]
Comment (vitlibar) [reply src/Storages/System/StorageSystemColumns.cpp:95]: Or there can be even more simple solution without changing User.h - just add `GRANT ALL ON system.*` in `parseUser()` (UsersConfigAccessStorage.cpp)

[the remark this answers is not in the document]
Comment (pufit) [reply src/Storages/System/StorageSystemColumns.cpp:95]: If I understand correctly, `access->isGranted(AccessType::SHOW_TABLES)` is equivalent to  `GRANT SHOW TABLES ON *.*`. So if a user doesn't have access to all tables, each table will be checked separately later, and not accessible tables will be removed from the results. 
The problem I'm trying to solve - we want to create a user, which can read all data from system tables, but it won't have access other than `system.*`.

[the remark this answers is not in the document]
Comment (vitlibar) [reply src/Storages/System/StorageSystemColumns.cpp:95]: `parseUser()` can also add implicit grant `GRANT SHOW TABLES ON *.*`

**Comment by vitlibar** [comment]
More general idea - we could implement a way to specify grants inside users.xml:

```
<users>
  <user_name>
    <grants>GRANT SELECT * ON system.*</grants>
    <grants>GRANT SHOW ON *.*</grants>
  </user_name>
</users>
```

**Comment by pufit** [comment]
Closed in favor of https://github.com/ClickHouse/ClickHouse/pull/49381

**Comment by zidane-wcp** [comment]
> 

I set it up, but it doesn't take effect

**Pull request closed without merging**